picocrt/machine/arm/crt0.c

```diff
@@ -255,6 +255,14 @@ _cstart(void)
 	__asm__("vmsr fpexc, %0" : : "r" (0x40000000));
 #endif
 
+/* Set up exception table base address (register VBAR_ELx).
+   Architectures earlier than v7 have the base address fixed. */
+#if __ARM_ARCH >= 7
+    uint32_t vbar;
+    __asm__("adr %0, __vector_table" : "=r"(vbar) : :);
```

Comment:
Could this have just been `__asm__("mcr P15, #0, %0, c12, c0, 0" : : "r" (__vector_table))? Let the compiler move the value into a register all on its own?